src/core/shell.rs

```diff
@@ -1,12 +1,18 @@
 #![allow(dead_code)]
 
+use anstream::{AutoStream, ColorChoice};
 use anyhow::Context;
 use clap::builder::styling::Style;
 use clap_cargo::style::{ERROR, HEADER, NOTE, WARN};
-use std::io::Write;
+use std::{
+    io::{Stderr, Write},
+    sync::OnceLock,
+};
 
 use crate::CargoResult;
 
+static COLOR_CHOICE: OnceLock<ColorChoice> = OnceLock::new();
```

Comment:
See https://docs.rs/colorchoice/latest/colorchoice/

`anstream` already reads from it so you don't need `fn stderr()`